docs/upgrade.md

```diff
@@ -92,12 +92,13 @@ process, for example:
 
 ## Changes to the account data replication streams
 
-Synapse has changed the format of the account data replication streams (between
-workers). This is a forwards- and backwards-incompatible change: v1.75 workers
-cannot process account data replicated by v1.76 workers, and vice versa.
+Synapse has changed the format of the account data and devices replication
+streams (between workers). This is a forwards- and backwards-incompatible
+change: v1.75 workers cannot process account data replicated by v1.76 workers,
+and vice versa.
```

Comment:
While a rolling upgrade is in progress, is there a risk of data loss?

When handling replication from old workers, new workers will raise exceptions and skip over replication messages, and vice versa.

Reply:
It should be fine. Since the format has changed workers will basically ignore *all* updates of the stream that is in the wrong format (either too old or too new). Once the workers update they'll receive `POSITION` commands and will then notice that they're missing data and re-request it.

Specifically, when a new server comes online it sends out new `POSITION` commands for all streams it is publishing, and all other workers also send it `POSITION` commands of all streams they're publishing. Receiving a `POSITION` command prompts the instance to pull any missing data from the DB and process that.